### PR TITLE
Feature/hierarchy not expanded

### PR DESCRIPTION
**Pull request by JorxSS**
Now the trees in the hierarchy and the project explorer are collapsed by default.

Hierarchy behaviour:
* When selecting from the scene, the GO in the hierarchy will be expanded and marked the selected one.
* If you select another GO the previous one will remain expanded.

Project Explorer behaviour:
* When selecting a folder you need to do double-click as usual, but to expand a folder that have sub-folder on it you need to use the arrow at the left side of the name.

In this PR you will find code from Animations, that's because I merged #166 because I was using that branch in the Game repo. My changes are the files PanelHierarchy and PanelProjectExplorer.

KNOWN BUG:
If you try to collapse a GO in the hierarchy that one of his childrens is the current selected you will need to make double click instead of just one click.

### CODE REVIEW

File: Engine/EditorUI/Panel/PanelHierarchy.h [hunk 1]
```diff
@@ -16,6 +16,8 @@ class PanelHierarchy : public Panel
 
 	std::string GetNextGameObjectName();
 	int GetNextBranch();
+	bool IsOneOfMyChildrens(GameObject * game_object);
+
```
Comment (marcorod94) [inline Engine/EditorUI/Panel/PanelHierarchy.h]: Code style

File: Engine/EditorUI/Panel/PanelHierarchy.h [hunk 2]
```diff
@@ -16,6 +16,8 @@ class PanelHierarchy : public Panel
...
 
 private:
 	void ShowGameObjectHierarchy(GameObject *game_object);
```
Comment (marcorod94) [inline Engine/EditorUI/Panel/PanelHierarchy.h]: Code style

**Review by marcorod94 — APPROVED** [review]
Really nice job!

File: Engine/EditorUI/Panel/PanelHierarchy.cpp [hunk 3]
```diff
@@ -360,4 +371,23 @@ std::string PanelHierarchy::GetNextGameObjectName()
...
+		if (!found && go->children.size() > 0)
+		{
+			found = IsOneOfMyChildrens(go);
+		}
+		if(found)
+		{
+			break;
+		}
+	}
+	return found;
```
Comment (Dartemiss) [inline Engine/EditorUI/Panel/PanelHierarchy.cpp]: Const mi pana

File: Engine/EditorUI/Panel/PanelProjectExplorer.cpp [hunk 4]
```diff
@@ -530,4 +541,22 @@ void PanelProjectExplorer::FilesDrop() const
...
+		if (!found && path->children.size() > 0)
+		{
+			found = IsOneOfMyChildrens(path);
+		}
+		if (found)
+		{
+			break;
+		}
+	}
+	return found;
```
Comment (Dartemiss) [inline Engine/EditorUI/Panel/PanelProjectExplorer.cpp]: Const again mi pana

**Review by Dartemiss — APPROVED** [review]
Works fine GJ

**Pull request merged**